src/main/java/com/epam/rd/autotasks/springemployeecatalog/repository/EmployeeRepositoryImpl.java

```diff
@@ -0,0 +1,119 @@
+import static com.epam.rd.autotasks.springemployeecatalog.constant.RepositoryConstants.BY_DEPARTMENT_ID;
+import static com.epam.rd.autotasks.springemployeecatalog.constant.RepositoryConstants.BY_DEPARTMENT_NAME;
+import static com.epam.rd.autotasks.springemployeecatalog.constant.RepositoryConstants.BY_ID;
+import static com.epam.rd.autotasks.springemployeecatalog.constant.RepositoryConstants.BY_MANAGER;
+import static com.epam.rd.autotasks.springemployeecatalog.constant.RepositoryConstants.DIGIT_REGEX;
+import static com.epam.rd.autotasks.springemployeecatalog.constant.RepositoryConstants.EMPLOYEE_BY_ID_SELECT;
+import static com.epam.rd.autotasks.springemployeecatalog.constant.RepositoryConstants.HIREDATE_CASE;
+import static com.epam.rd.autotasks.springemployeecatalog.constant.RepositoryConstants.LASTNAME_CASE;
+import static com.epam.rd.autotasks.springemployeecatalog.constant.RepositoryConstants.LIMIT_OFFSET;
+import static com.epam.rd.autotasks.springemployeecatalog.constant.RepositoryConstants.POSITION_CASE;
+import static com.epam.rd.autotasks.springemployeecatalog.util.StatementUtils.getPreparedStatement;
+
+@Repository
+public class EmployeeRepositoryImpl implements EmployeeRepository{
+    public static final String ORDER_BY_LASTNAME = " ORDER BY LASTNAME_e";
+    public static final String ORDER_BY_HIREDATE = " ORDER BY HIREDATE_e";
+    public static final String ORDER_BY_POSITION = " ORDER BY POSITION_e";
+    public static final String ORDER_BY_SALARY = " ORDER BY SALARY_e";
+    private final JdbcTemplate jdbcTemplate;
+    private final ExtractorFactory factory;
+
+    @Autowired
+    public EmployeeRepositoryImpl(JdbcTemplate jdbcTemplate, ExtractorFactoryImpl factory) {
+        this.jdbcTemplate = jdbcTemplate;
+        this.factory = factory;
+    }
+
+    public Employee getEmployeeById(Long id, boolean withChain) {
+        String finalQuery;
+        if (withChain) {
+            finalQuery = EMPLOYEE_BY_ID_SELECT;
+        } else {
+            finalQuery = ALL_EMPLOYEES + BY_ID;
+        }
+        return jdbcTemplate.query(connection -> {
+                    PreparedStatement statement = getPreparedStatement(finalQuery, connection);
+                    statement.setLong(1, id);
+                    return statement;
+                }, factory.getExtractor(withChain))
+                .get(0);
+    }
+
+    public List<Employee> getAllEmployees(Long page, Long size, String sort) {
+        if (page != null) {
+            String finalQuery = ALL_EMPLOYEES + getOrder(sort) + LIMIT_OFFSET;
+            int offset = Math.toIntExact(page * size);
+            return jdbcTemplate.query(connection -> {
+                PreparedStatement statement = getPreparedStatement(finalQuery, connection);
+                statement.setInt(1, Math.toIntExact(size));
+                statement.setInt(2, offset);
+                return statement;
+            }, factory.getExtractor(false));
```

Comment:
It is more readable
Object[] params = {Math.toIntExact(size), offset};
return jdbcTemplate.query(finalQuery, params, factory.getExtractor(false));